### PR TITLE
jfrog-cli: init at 1.43.2

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by invokesus** [comment]
I couldn't find a clean way of passing `-o jfrog` to `go build`, hence the `postInstall`. Let me know if there's a better way. :)

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  postInstall = ''
+    mv $out/bin/jfrog-cli $out/bin/jfrog
+  '';
+
+  meta = with lib; {
+    description =
+      "JFrog CLI is a client that provides a simple interface that automates access to the JFrog products.";
+    homepage = "https://github.com/jfrog/jfrog-cli";
+    license = licenses.asl20;
+    maintainers = [ maintainers.deshaw ];
```
Comment (grahamc) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:35]: A little fixup: the teams is part of a different variable, `teams`:

```suggestion
    maintainers = teams.deshaw.members;
```

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  # Most of them seem to require network access.
+  doCheck = false;
+
+  postInstall = ''
+    mv $out/bin/jfrog-cli $out/bin/jfrog
+  '';
+
+  meta = with lib; {
+    description =
+      "JFrog CLI is a client that provides a simple interface that automates access to the JFrog products.";
```
Comment (grahamc) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:32]: Omit the period at the end of descriptions: https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes

```suggestion
      "A client that provides a simple interface that automates access to the JFrog products";
```

**Comment by invokesus** [comment]
Thanks for the review. Made the changes.

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "0gc8a4cpzcpa3mrqzhba4xx0316z030q71bpxlslivxg5xpvgq0v";
+  };
+
+  CGO_ENABLED = 0;
+
+  vendorSha256 = "1imlc8vmhph9gy4c9xyr2czfyh42p9crh37h5axc5939lnn4x9z6";
+
+  # Extracted from build/build.sh.
+  buildFlagsArray = [
+    "-ldflags=-w -extldflags '-static'"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:20]:
```suggestion
    "-ldflags=-s -w -extldflags '-static'"
```

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  ];
+
+  # Most of them seem to require network access.
+  doCheck = false;
+
+  postInstall = ''
+    mv $out/bin/jfrog-cli $out/bin/jfrog
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:27]: Maybe we want to symlink the binary incase any guides reference the other name.

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  '';
+
+  meta = with lib; {
+    description = "A client that provides a simple interface that automates access to the JFrog products";
+    homepage = "https://github.com/jfrog/jfrog-cli";
+    license = licenses.asl20;
+    maintainers = [ teams.deshaw.members ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:35]:
```suggestion
```

Set by buildGoModule.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110339` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>jfrog-cli</li>
</ul>
</details>

File: pkgs/development/tools/misc/jfrog-cli/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/misc/jfrog-cli/default.nix:20]: It tells go to strip things right away. -w removes dwarf.

**Comment by invokesus** [comment]
Thanks for the review, @SuperSandro2000! Made the changes.

File: pkgs/development/tools/misc/jfrog-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  buildFlagsArray = [
+    "-ldflags=-s -w -extldflags '-static'"
+  ];
+
+  # Most of them seem to require network access.
+  doCheck = false;
+
+  postInstall = ''
+    mv $out/bin/jfrog-cli $out/bin/jfrog
+    ln -s $out/bin/jfrog $out/bin/jfrog-cli
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/jfrog-cli/default.nix:28]:
```suggestion
    ln -s $out/bin/jfrog-cli $out/bin/jfrog
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**